packages/components/src/shortcuts/ShortcutRegistry.ts

```diff
@@ -0,0 +1,64 @@
+import Shortcut, { KeyState } from './Shortcut';
+
+export default class ShortcutRegistry {
+  static readonly shortcutMap = new Map<string, Shortcut>();
+
+  static readonly shortcutsByCategory = new Map<string, Shortcut[]>();
+
+  /**
+   * Creates a Shortcut and adds it to the registry
+   * @param params The constructor params for the {@link Shortcut}
+   * @param category The category this will show up under in the settings menu. If omitted, shortcut will be hidden in settings
+   * @returns The created Shortcut
+   */
+  static createAndAdd(
+    params: ConstructorParameters<typeof Shortcut>[0]
+  ): Shortcut {
+    const shortcut = new Shortcut(params);
+    ShortcutRegistry.add(shortcut);
+    return shortcut;
+  }
+
+  /**
+   * Adds a shortcut to the registry. Throws if a shortcut with the same ID already exists
+   * @param shortcut Shortcut to add to the registry
+   */
+  static add(shortcut: Shortcut): void {
+    if (ShortcutRegistry.shortcutMap.has(shortcut.id)) {
+      throw new Error(
+        `Skipping attempt to add duplicate shortcut ID to registry: ${shortcut.id}`
+      );
+    }
+
+    const category = shortcut.id.split('.')[0];
+
+    ShortcutRegistry.shortcutMap.set(shortcut.id, shortcut);
+    if (ShortcutRegistry.shortcutsByCategory.has(category)) {
```

Comment:
I don't think you need to have a `shortcutsByCategory` map in the registry itself, unless we enforce something like a shortcut can't conflict with something in the same category.
Even then - we should probably let the user set a conflicting shortcut, and just inform them that there's an existing shortcut for that combo.

Reply:
The thought was to only warn about conflicting shortcuts within categories. Should be easy enough to just calculate the category list each time instead of keeping a map of it.